prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -562,8 +562,14 @@ class ParserSpec extends AnyFunSpec with Matchers {
 
-      "(metric1 + metric2) * (metric3 + metric4)" -> "BinaryJoin(BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric1))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric2))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),MUL,OneToOne,BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric3))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric4))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),List(),List(),List())"
-
+      "(metric1 + metric2) * (metric3 + metric4)" -> "BinaryJoin(BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric1))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric2))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),MUL,OneToOne,BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric3))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric4))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),List(),List(),List())",
+      "foo[5m:1m]" -> "PeriodicSeries(RawSeries(IntervalSelector(1524855688000,1524855988000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(300000),None),1524855688000,60000,1524855988000,None)",
+      "rate(foo[3m])[5m:1m]" -> "PeriodicSeriesWithWindowing(RawSeries(IntervalSelector(1524855688000,1524855988000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(180000),None),1524855688000,60000,1524855988000,180000,Rate,false,List(),None)"
```

Comment:
There are four kinds of test cases here:
1. subquery expression on raw metric like `foo[5m:1m]`
2. subquery expression on transformed metric like `rate(foo[3m])[5m:1m]`
3. subquery expression with range function `max_over_time(sum(rate(foo[5m]))[5m:1m])`
4. subquery expression with range function, followed by further transformation `sum(rate(foo[5m])[5m:1m])`

Priority of cases is: 3 & 4, followed by 2 & 1

Can we make the parser work for cases 3 & 4 first ?

Reply:
We should also get the resulting logical plan right in the PR. Let's create the SubqueryRangeFunction logical plan and use it.

Reply:
1, 2, 3, 4 without aggregates should work now. The new logical plan is there as well.